c-dependencies/js-compute-runtime/builtins/backend.cpp

```diff
@@ -17,6 +19,617 @@
+    // or TLS, they only affect the list of available cipher suites.
+    aliases.insert({SSL_PROTO_TLSv1_2, filterByProtocol(all, std::set{Protocol::TLSv1_2})});
+    auto tlsv1 = filterByProtocol(all, std::set{Protocol::TLSv1});
+    aliases.insert({SSL_PROTO_TLSv1_0, tlsv1});
+    aliases.insert({SSL_PROTO_TLSv1, tlsv1});
+    aliases.insert({SSL_PROTO_SSLv3, filterByProtocol(all, std::set{Protocol::SSLv3})});
+
+    // cipher suites using 128 bit AES.
+    auto aes128 = filterByEncryption(all, std::set{Encryption::AES128, Encryption::AES128GCM});
+    aliases.insert({AES128, aes128});
+    // cipher suites using 256 bit AES.
+    auto aes256 = filterByEncryption(all, std::set{Encryption::AES256, Encryption::AES256GCM});
+    aliases.insert({AES256, aes256});
+    // cipher suites using either 128 or 256 bit AES.
+    auto aes(aes128);
+    aes.insert(aes.end(), aes256.begin(), aes256.end());
+    aliases.insert({AES, aes});
+
+    // AES in Galois Counter Mode (GCM).
+    aliases.insert(
+        {AESGCM, filterByEncryption(all, std::set{Encryption::AES128GCM, Encryption::AES256GCM})});
+
+    // Cipher suites using ChaCha20.
+    aliases.insert({CHACHA20, filterByEncryption(all, std::set{Encryption::CHACHA20POLY1305})});
+
+    // Cipher suites using triple DES.
+    auto triple_des = filterByEncryption(all, std::set{Encryption::TRIPLE_DES});
+    aliases.insert({TRIPLE_DES, triple_des});
+
+    // Cipher suites using SHA1.
+    auto sha1 = filterByMessageDigest(all, std::set{MessageDigest::SHA1});
+    aliases.insert({SHA1, sha1});
+    aliases.insert({SHA, sha1});
+    // Cipher suites using SHA256.
+    aliases.insert({SHA256, filterByMessageDigest(all, std::set{MessageDigest::SHA256})});
+    // Cipher suites using SHA384.
+    aliases.insert({SHA384, filterByMessageDigest(all, std::set{MessageDigest::SHA384})});
+
+    // COMPLEMENTOFDEFAULT:
+    // The ciphers included in ALL, but not enabled by default. Currently this includes all RC4 and
+    // anonymous ciphers. Note that this rule does not cover eNULL, which is not included by ALL
+    // (use COMPLEMENTOFALL if necessary). Note that RC4 based cipher suites are not supported by
+    // Fastly and the only supported anonymous ciphers are `ecdh` and `triple_des`.
+    auto complementOfDefault(ecdh);
+    complementOfDefault.insert(complementOfDefault.end(), triple_des.begin(), triple_des.end());
+    aliases.insert({COMPLEMENTOFDEFAULT, complementOfDefault});
+
+    // The content of the default list is determined at compile time and normally corresponds to
+    // ALL:!COMPLEMENTOFDEFAULT:!eNULL.
+    aliases.insert({DEFAULT, parse("ALL:!COMPLEMENTOFDEFAULT:!eNULL")});
+  }
+
+  std::vector<Cipher> parse(std::string_view expression) {
+    /**
+     * All ciphers by their openssl alias name.
+     */
+    auto elements = splitCipherSuiteString(std::string(expression));
+    std::vector<Cipher> ciphers;
+    std::vector<Cipher> removedCiphers;
+    for (auto element : elements) {
```

Comment:
This string is getting copied for every iteration currently.
```suggestion
    for (auto &element : elements) {
```